lib/elixir/lib/kernel.ex

```diff
@@ -3087,6 +3087,16 @@ defmodule Kernel do
           {expanded, nil}
       end
 
+    # We do this so that the block is not tail-call optimized and stacktraces
+    # are not messed up. Basically, we just insert something between the return
+    # value of the block and what is returned by defmodule. Using just ":ok" or
+    # similar doesn't work because it's likely optimized away by the compiler.
+    block = quote do
+      result = unquote(block)
+      (fn -> :ok end).()
```

Comment:
We should not use anonymous functions because there is a part of the compiler that cannot be optimized or discarded if it sees anonymous functions in the module body (iirc). I would do `:elixir.noop()`. :)